src/opensynth/models/faraday/gaussian_mixture/gmm_init.py

```diff
@@ -14,23 +14,23 @@ def initialise_gmm_params(
     reg_covar: float = 1e-6,
 ) -> gmm_model.GMMInitParams:
     """
-    Initialise Gaussian Mixture Parameters. This works
-    by only initialising on the first batch of the data
-    using K-means, and porting SKLearn's implementation
-    of computing cholesky precision and covariances.
+    Initialise Gaussian Mixture Parameters.
+    Component means and component assignments are initialised using K-means,
+      and porting SKLearn's implementation of computing cholesky precision and
+      covariances.
 
     Args:
-        X (np.array): Input data
+        X (torch.Tensor): Input data
         n_components (int): Number of components
         reg_covar (float): Regularisation for covariance matrix
 
     Returns:
         dict[str, torch.Tensor]: GMM params
     """
 
-    # Use data from first batch to initialise centroids
-    # If data is shuffled, this should represent the full dataset
     X = gmm_utils.encode_data(data, vae_module)
+    if "weights" in list(data.keys()):
```

Comment:
♻️ Same comment about `weights`:

- either ask users explicitly for the name of the column, or enumerate it in `TrainingData` as a NotRequired key.